### PR TITLE
Add a test for WARNINGS.md

### PR DESCRIPTION
**Pull request by vladmos**
All warnings contain a link to `https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#<category_name>`, the new test ensures that all registered warnings are documented in the file (at least that there's a link in the header and an anchor for each warning).

### CODE REVIEW

File: warn/documentation_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	contents := string(data)
+	for _, warning := range AllWarnings {
+		link := fmt.Sprintf("  * [%s](#%s)", warning, warning)
+		if !strings.Contains(contents, link) {
+			t.Errorf("No link found for the warning %q in WARNINGS.md, is it documented?", warning)
```
Comment (laurentlb) [inline warn/documentation_test.go]: nit. the error message should include the pattern you match (`link`).

File: warn/documentation_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		}
+
+		anchor := fmt.Sprintf(`<a name="%s"></a>`, warning)
+		if !strings.Contains(contents, anchor) {
+			t.Errorf("No anchor found for the warning %q in WARNINGS.md, is it documented?", warning)
```
Comment (laurentlb) [inline warn/documentation_test.go]: here, include `anchor`

File: warn/documentation_test.go (hunk 1 above)
Comment (vladmos) [reply warn/documentation_test.go]: Done.

File: warn/documentation_test.go (hunk 2 above)
Comment (vladmos) [reply warn/documentation_test.go]: Done.

**Pull request merged**